scripts/r2s.py

```
@@ -163,6 +165,10 @@ def step2():
     config.read(config_filename)
     structured = config.getboolean('general', 'structured')
     sub_voxel = config.getboolean('general', 'sub_voxel')
+    try:
```

Let's set the `fallback` parameter of [`ConfigParser.get()`](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.get) to `'MCNP'` here. It will allow us to combine these 4 lines into one.

---

Also, let's make sure that the value of `transport_code` is one of `OpenMC` or `MCNP` here so we can fail early and raise an exception with a clear error message if the value is invalid.

---

As we do not need to add filetype to the `source.h5m` when OpenMC is used. We do not this `transport_code` too.  It is removed in https://github.com/pyne/pyne/pull/1355/commits/c09fe65f12c7a537ba0fe747c163e9b7808f0a5b.